convert2rhel/checks.py

```diff
@@ -772,20 +772,40 @@ def is_loaded_kernel_latest():
     # decide wether to bail out or output a warning (only if the user used the
     # special environment variable for it.
     if not packages:
-        unsupported_skip = os.environ.get("CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK", None)
-        if not unsupported_skip:
-            logger.critical(
-                "Could not find any %s from repositories to compare against the loaded kernel.\n"
-                "Please, check if you have any vendor repositories enabled to proceed with the conversion.\n"
-                "If you wish to ignore this message, set the environment variable "
-                "'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' to 1." % package_to_check
+        # Repoquery failed to detected any kernel or kernel-core packages in it's repositories
+        # we allow the user to provide a environment variable to override the functionality and proceed
+        # with the conversion, otherwise, we just throw an critical logging to them.
+        allow_older_envvar_names = (
+            "CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK",
+            "CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK",
+        )
+
+        # This check is to see which environment variable is set, To allow users in the next version
+        # to adjust their environmental variable names. This check will be removed in the future and
+        # will only have the 'CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK' environment variable
+        if any(envvar in os.environ for envvar in allow_older_envvar_names):
+            if "CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK" in os.environ:
+                logger.warning(
+                    "You are using the deprecated 'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK'"
+                    " environment variable.  Please switch to 'CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK'"
+                    " instead."
+                )
+
+            logger.warning(
+                "Detected 'CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK' environment variable, we will skip "
+                "the %s comparison.\nBeware, this could leave your system in a broken state." % package_to_check
```

Comment:
This is a bit confusing in my opinion.
First, we warn the user to not use the deprecated envar and inform of using the new one, then we warn about detecting the new one.
WDYT @r0x0d @bocekm @SpyTec ?